### DISCUSSION TITLE
[BUG] Incorrect time after uploading

### DISCUSSION THREAD

**Discussion by dm3010** [immich-app/immich#12322] [Q&A]
### The bug

![image](https://github.com/immich-app/immich/assets/88131985/eeac2791-43e5-47b6-aafd-2fab464bdcb5)
![image](https://github.com/immich-app/immich/assets/88131985/6fbbd892-aaff-4eec-8afe-fd4d81568915)

These are two photos from the same phone, taken 3 hours apart in time
Info about before uploading
Time **15:00** (1 photo) and **18:01** (2 photo) - **Correct** all

![image](https://github.com/immich-app/immich/assets/88131985/95fc9fdf-6957-448a-b2ac-3de34fb5ee2e)
![image](https://github.com/immich-app/immich/assets/88131985/7832a974-9c0c-438c-ab39-589101b605c6)

After uploading from mobile app, one of the photos has the wrong time
Time **18:00** (1 photo) **wrong!** and **18:01** (2 photo) - **correct**

I've tried uploading them via the web interface as well, and the problem is the same

The problem seems to be due to timezone, as the time shift on the wrong photo is the same as +3 hours (My timezone GMT+3)

What surprises me the most is that these photos were taken on the same day with the same phone

### The OS that Immich Server is running on

Docker

### Version of Immich Server

v1.85

### Version of Immich Mobile App

v1.85

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
*
```

### Your .env content

```Shell
*
```

### Reproduction steps

```bash
1.
2.
3.
...
```

**Comment by andyxpert** [comment]
do you have the timezone set correctly ?
https://immich.app/docs/install/environment-variables#general - look for TZ env variable

**Comment by dm3010** [comment]
I tried adding this option (TZ="GMT+3"), and got an increase in first photo time by another three hours
First photo time is set to 21:00

**Comment by misunonozomi** [comment]
I have the same problem here, the time was added 8 hours (cause I'm in GMT+8)
This happened on the iOS screenshot.
I tried added ```TZ="GMT+8"``` but that made no change at all. 
[Original File Here](https://github.com/immich-app/immich/assets/45501123/1fab74f3-86fe-446d-90cf-0300149ad412)
![IMG_1460](https://github.com/immich-app/immich/assets/45501123/7adb2296-97fe-4746-8c13-2f541a960902)

**Comment by misunonozomi** [comment]
Also a related bug.
Though mobile version can. Immich webui won't list photos "from the future".

**Comment by MDsza** [comment]
I am loving Immich more and more... but same problem here (in Switzerland = GMT/UTC+1): **Only iOS Screenshots are shown with wrong timestamp =  +1 hour in the future after uploading to Immich** what results in a wrong order of the images on my iPhone AND the WebUI, too. Images shot on the iPhone are shown with the right time stamp... hmmm. Any solution?

**Comment by DJPapa66** [comment]
Hello,

I have the same problem and don't understand why the recording date isn't just taken. E.G. the photo was taken at 15:52:37 and not 16:52:37. That's it. Why does something have to be calculated? I'm "lucky" that it's only +1 hour (Berlin) for me. I could deal with that if necessary. However, I don't understand it.
It would be great if I could choose how to import - maybe in the settings (with or without calculation).

**Comment by BjGoCraft** [comment]
Same problem here: Two photos were taken 3 seconds apart. When looking at the metadata locally on my PC, the time on both is right. When uploading both pictures to immich via the web interface, one has the correct time stamp and one is suddenly one hour later (GMT +1). 
Edit: I just found out that the picture with the right time has location data while the other hasn't.

**Comment by alextran1502** [comment]
@BjGoCraft can you share those teo photos for troubleshooting?

Saw your edit, that explains it

**Comment by BjGoCraft** [comment]
> @BjGoCraft can you share those teo photos for troubleshooting?
> 
> Saw your edit, that explains it

I did send you an e-mail, I don't want to publish my face :)

**Comment by ygaeon** [comment]
Can confirm I experience this as well ..

(tried to attach photos here but unable to)

Photo date/timestamp on phone is `2024-04-15 15:37:57`
Phone timezone is CET (GMT +2).

Once uploaded to Immich server, photo date/timestamp becomes `2024-04-15 17:37 GMT+02:00`

....

While researching a bit, discovered that `/etc/localtime` was symlinked to `/usr/share/zoneinfo/Etc/UTC` even though I created the container with `--volume /etc/localtime:/etc/localtime:ro` and the host's localtime is linked to CEST.

Do you set this `/etc/localtime` from somewhere else starting up?

Please advice ..

**Comment by BjGoCraft** [comment]
My `/etc/localtime` is symlinked to `/usr/share/zoneinfo/Europe/Berlin` so that is right. 
In my docker-compose.yml und the immich-microservices > volumes: > there is - /etc/localtime:/etc/localtime:ro
so all time zones are set to local time. The timezone isn't specified anywhere else. I didn't create the container with any flags but as described in the [docs](https://immich.app/docs/install/docker-compose). The problem @dm3010 is experiencing too seems to be linked to if the photo has location data or not

**Comment by semiligneous** [comment]
I get this too. Seems quite simple to reproduce:

- Docker compose installation, no customisation.
- host tz as something ahead of utc by a few hours
- Leave /etc/localtime mounted from host
- take a screenshot on iOS and immediately upload it via the app
- take a photo on iOS and immediately upload it via the app

Observe that the photo correctly appears with date/time in local time showing in timeline as expected 

Observe that the screen shot shows in timeline 'in the future'- if your utc tz offset is large and you took the screenshot late enough in the day, it will even show under tomorrow's date in the timeline 

The date itself is off by whatever the utc offset happens to be.

Seems to be impossible to fix (edit: through config files) since it affects only certain media - in this case screenshots but as others have said, possibly also media without a location?
I'm

**Reply by jrasm91** [reply]
This has been fixed in #12612

**Reply by ygaeon** [reply]
Not fixed, see #9331

**Reply by alextran1502** [reply]
@ygaeon the pr that fixed the issue has been merged but not yet released

**Reply by ygaeon** [reply]
Ah, my bad @alextran1502 .. checking #12612 -> #8243 .. which is not same issue as #9331 ..
.. but this discussion / issue "Incorrect time after uploading" is still an issue .. so suggesting to keep this discussion / issue open still ..

**Comment by jrasm91** [comment]
Does anyone have an image they can share with me that when uploaded to immich has the wrong date? You can upload it here (in a zip) or send it to me via email or discord and I can take a closer look.

**Comment by sirweazel** [comment]
I've been having this issue, but only noticed it in the evening times.  Today to look a few minutes to look into it.  After uploading some images this evening  from a trip i took today.  I logged into the console of each container and used the Linux "date" command.  The database container immich_postgres, appears to have the wrong time. The other containers appear to have the correct time that is used in the environment variable. For me it is "America/New_York". Do you think this is the issue?
I grabbed various screenshots (including the time i grabbed them) to show that the metatdata is identified as the correct time, and timeline shows as "yesterday" along with screenshots from 2 containers where 1 has the correct time and the other does not.  If you need something else let me know. 


![image](https://github.com/user-attachments/assets/0ca38c0e-b237-47dc-857a-f10826d2642c)

![image](https://github.com/user-attachments/assets/ce1a67ef-f499-4c17-85b2-32837a0531d3)

![image](https://github.com/user-attachments/assets/3bdea644-559d-46b4-81c3-d7c1325cca4a)
![image](https://github.com/user-attachments/assets/26043425-def0-47ce-9d4c-6e167be82167)
![image](https://github.com/user-attachments/assets/882272bd-748f-4ded-a29a-f20e1c262639)

**Comment by ygaeon** [comment]
@bo0tzz Why was this discussion closed? This is still an issue .. and an active bug is still in progress #9331 with little to no attention.

**Reply by bo0tzz** [reply]
I was going through a tidy and saw this comment https://github.com/immich-app/immich/discussions/12322#discussioncomment-10632601.

**Reply by ygaeon** [reply]
OK, added comment above to highlight #9331 .. as this issue is not fixed. Still having wrong dates after images been uploaded (videos seem to have correct timestamp though)

**Reply by ygaeon** [reply]
@bo0tzz .. perhaps keep this discussion / issue open?

**Reply by bo0tzz** [reply]
We've decided to collect all the time related issues in  #12650

**Reply by ygaeon** [reply]
Ah, a newly created discussion thread .. thanks .. wouldn't have known otherwise :-)

**Comment by sirweazel** [comment]
Do you think this is becasue of the DB container not using the correct timezone?  Or is it supposed to be using UTC time only? I'm not a DB guy.

**Reply by ygaeon** [reply]
@sirweazel check this comment https://github.com/immich-app/immich/issues/9331#issuecomment-2284195781 .. your DB timezone may not be the root cause .. /shrug/
